File: src/09.tree/23.257.go
package tree

import "strconv"

func BinaryTreePaths(root *TreeNode) []string {
	var dfs func(*TreeNode, string)
	ans := []string{}

	dfs = func(tn *TreeNode, s string) {
		if tn == nil {
			return
		}

		if s == "" {
			s += strconv.Itoa(tn.Val)
		} else {
			s += "->" + strconv.Itoa(tn.Val)
		}

		if tn.Left == nil && tn.Right == nil {
			ans = append(ans, s)
			return
		}

		dfs(tn.Left, s)
		dfs(tn.Right, s)
	}

	dfs(root, "")

	return ans
}

File: src/09.tree/23.257.ts
import { TreeNode } from "./01.100";

export function binaryTreePaths(root: TreeNode | null): string[] {
    // BFS
    const ans: string[] = [];
    const dfs = (node: TreeNode | null, cur: string) => {
        if (!node) {
            return;
        }

        if (!cur) {
            cur += node.val + "";
        } else {
            cur += "->" + node.val;
        }

        if (!node.left && !node.right) {
            ans.push(cur);
            return;
        }

        dfs(node.left, cur);
        dfs(node.right, cur);
    };

    dfs(root, "");

    return ans;
}
